modules/xmpp/xmpp.js

```diff
@@ -810,7 +810,7 @@ export default class XMPP extends Listenable {
     disconnect(ev) {
         if (this.disconnectInProgress) {
             return this.disconnectInProgress;
-        } else if (!this.connection) {
+        } else if (!this.connection || !this.connection.connected) {
```

Comment:
I'm wondering is it possible to call connect and without waiting for result call disconnect. 

It looks to me that after your change we won't call disconnect on Strophe at all and before we would have called it. Is this correct?

Do we care about this use case? The previous version sounds more correct to me.

Reply:
> I'm wondering is it possible to call connect and without waiting for result call disconnect.

Nope, we have see a lot of problems because of that. Where we connect quickly and disconnect event is fired and everythng messed up we think we are connected, the library thinks it is disocnnected.



> It looks to me that after your change we won't call disconnect on Strophe at all and before we would have called it. Is this correct?

Well if it is not connected we will skip. Well maybe it sounds correct but there was a particular problem that was fixed with this.

Reply:
And this is exactly with the visitors. As we always call disconnect there and this promise was never resolving and everything hangs.